File: aemet/SergioDeIscar-aemet-java/SergioDeIscar-aemet-java/src/main/java/org/example/config/AppConfig.java
package org.example.config;

import java.io.File;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.Properties;

public class AppConfig {
    // Singleton
    private static AppConfig instance = null;
    public static AppConfig getInstance() {
        initStorage();
        loadProperties();
        if (instance == null) {
            instance = new AppConfig();
        }
        return instance;
    }

    private String APP_NAME = "Herencia DTO";
    private String APP_VERSION = "1.0.0";
    private static String APP_AUTHOR;
    private static String APP_DATA;
    private static String LOCAL_PATH = System.getProperty("user.dir") + File.separator;

    private static void initStorage(){
        try {
            Files.createDirectories(Paths.get("data"));
        }catch (Exception e){
            System.out.println("Error al crear el directorio de almacenamiento");
        }
    }

    private static void loadProperties(){
        Properties properties = new Properties();
        try {
            properties.load(AppConfig.class.getResourceAsStream("/config.properties"));
            APP_AUTHOR = properties.getProperty("app.author");
            APP_DATA = properties.getProperty("app.storage.dir");
            APP_DATA = LOCAL_PATH + APP_DATA;
        }catch (Exception e){
            System.out.println("Error al cargar las propiedades");
        }
    }

    public String getAppName() {
        return APP_NAME;
    }
    public String getAppVersion() {
        return APP_VERSION;
    }
    public String getAppAuthor() {
        return APP_AUTHOR;
    }
    public String getAppData() {
        return APP_DATA;
    }
}

File: buerguer-pig/EmmaFernandez-RestauranteKotlin/src/main/kotlin/config/AppConfig.kt
package config

import java.io.File
import java.nio.file.Files
import java.nio.file.Paths
import java.util.Properties

private val LOCAL_PATH = "${System.getProperty("user.dir")}${File.separator}"

object AppConfig {
    lateinit var APP_DATA: String
    lateinit var SERIALIZABLE_PATH: String
    lateinit var CSV_PATH: String

    init {
        loadProperties()
        initStorage()
    }

    private fun loadProperties() {
        val properties = Properties()
        properties.load(AppConfig::class.java.getResourceAsStream("/config.properties"))
        APP_DATA = properties.getProperty("app.data.dir")
        SERIALIZABLE_PATH = APP_DATA + File.separator + properties.getProperty("app.data.serializable.dir")
        CSV_PATH = APP_DATA + File.separator + properties.getProperty("app.data.csv.dir")
    }

    private fun initStorage() {
        Files.createDirectories(Paths.get(SERIALIZABLE_PATH))
        Files.createDirectories(Paths.get(CSV_PATH))
    }
}
